enterprise_reporting/models.py

```diff
@@ -0,0 +1,157 @@
+"""
+from __future__ import absolute_import, unicode_literals
+
+from django.core.exceptions import ValidationError
+from django.core.validators import MaxValueValidator, MinValueValidator
+from django.db import models
+from django.utils.crypto import get_random_string
+from django.utils.encoding import python_2_unicode_compatible
+from django.utils.translation import ugettext_lazy as _
+import datetime
+import pytz
+
+from model_utils.models import TimeStampedModel
+
+from enterprise.models import EnterpriseCustomer
+
+
+@python_2_unicode_compatible
+class EnterpriseCustomerReportingConfiguration(TimeStampedModel):
+    """
+    The Enterprise's configuration for sending automated data reports securely via email to the Enterprise Admin.
+    """
+
+    FREQUENCY_TYPE_DAILY = 'daily'
+    FREQUENCY_TYPE_MONTHLY = 'monthly'
+    FREQUENCY_TYPE_WEEKLY = 'weekly'
+
+    FREQUENCY_CHOICES = (
+        (FREQUENCY_TYPE_DAILY, FREQUENCY_TYPE_DAILY),
+        (FREQUENCY_TYPE_MONTHLY, FREQUENCY_TYPE_MONTHLY),
+        (FREQUENCY_TYPE_WEEKLY, FREQUENCY_TYPE_WEEKLY),
+    )
+
+    DAYS_OF_WEEK = (
+        (0, 'Monday'),
+        (1, 'Tuesday'),
+        (2, 'Wednesday'),
+        (3, 'Thursday'),
+        (4, 'Friday'),
+        (5, 'Saturday'),
+        (6, 'Sunday'),
+    )
+
+    enterprise_customer = models.OneToOneField(
+        EnterpriseCustomer, blank=False, null=False, verbose_name=_("Enterprise Customer")
+    )
+    active = models.BooleanField(blank=False, null=False, verbose_name=_("Active"))
+    email = models.EmailField(blank=False, null=False, verbose_name=_("Email"))
+    frequency = models.CharField(
+        max_length=20,
+        choices=FREQUENCY_CHOICES,
+        blank=False,
+        default=FREQUENCY_TYPE_MONTHLY,
+        verbose_name=_("Frequency"),
+        help_text=_("The frequency interval (daily, weekly, or monthly) that the report should be sent."),
+    )
+    day_of_month = models.SmallIntegerField(
+        blank=True,
+        null=True,
+        verbose_name=_("Day of Month"),
+        help_text=_("The day of the month to send the report. "
+                    "This field is required and only valid when the frequency is monthly."),
+        validators=[MinValueValidator(1), MaxValueValidator(31)]
+    )
+    day_of_week = models.SmallIntegerField(
+        blank=True,
+        null=True,
+        choices=DAYS_OF_WEEK,
+        verbose_name=_("Day of Week"),
+        help_text=_("The day of the week to send the report. "
+                    "This field is required and only valid when the frequency is weekly."),
+    )
+    hour_of_day = models.SmallIntegerField(
+        blank=False,
+        null=False,
+        verbose_name=_("Hour of Day"),
+        help_text=_("The hour of the day to send the report, in Eastern Standard Time (EST). "
+                    "This is required for all frequency settings."),
+        validators=[MinValueValidator(0), MaxValueValidator(23)]
+    )
+    password = models.CharField(
```

Comment:
What is this password _for_? I'm _very_ squeamish about having any sort of password stored in the database as plain text.

Reply:
Great question, open to implementing differently, especially while it's still easy to change. Essentially we need a password for encrypting the csv that can be shared with the customer separately. This job needs access to that field and it needs to be configured per enterprise customer.

Reply:
Could we do offline generation of a public/private keypair? Then, we can store the public key in the DB and use it for encryption operations, and provide the private key to the customer to safeguard.

Reply:
The implementation we agreed upon for this initial story was to send password protected zip files. I'm planning to use this package to do that: https://github.com/smihica/pyminizip
Maybe this is something we can talk about after standup tomorrow, because I agree this is not ideal but my impression from how we groomed this story was that we were not going to go the full public/private key route, but I would like to not store a password in the database.

Reply:
Gotcha; that makes sense. Yeah, I don't want to explode the scope too much - does @fredsmith have any thoughts on specific implementation?